sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+using System.Collections.Immutable;
+using csharp::SonarAnalyzer.Rules.CSharp;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class ParameterLoaderTests
```

Comment:
I am only testing the explicit part (i.e. no test of an invalid xml format, file not found...)

Reply:
Why do you mean by "explicit"?

Why don't you test for invalid xml/file not found? That would be useful to ensure we fail gracefully.

Reply:
What I mean is that the code is not making any check and so will fail with the default ways so I don't think it is worth testing these cases.